homeassistant/components/lcn/helpers.py

```diff
@@ -64,6 +65,7 @@ def get_device_connection(hass, address, config_entry):
 
 def get_resource(domain_name, domain_data):
     """Return the resource for the specified domain_data."""
+    assert domain_name in PLATFORMS
```

Comment:
`assert` should be avoided in use. It is OK to use with typing, but not functional.

Reason is that they are ignored when Python is running in an optimized mode.

Reply:
Thanks for the explanation! I reverted the changes and replaced the asserts by exceptions.